pkg/sql/internal.go

```diff
@@ -106,6 +117,47 @@ func MakeInternalExecutor(
 	}
 }
 
+// MakeInternalExecutorWithTxn creates an Internal Executor with txn related
+// information.
+func MakeInternalExecutorWithTxn(
```

Comment:
Given you end up accessing the internal executor as a pointer in the closure you return (it's a pointer method). I think you should give up on avoiding the allocation and call this `NewInternalExecutor...` and return a `*InternalExecutor` so that it's the same object.

Reply:
Done